File: Android/app/src/main/java/realm/io/realmpop/util/RandomNumberUtils.java
package realm.io.realmpop.util;

import java.util.Random;

public class RandomNumberUtils {

    private static Random rand = new Random(System.currentTimeMillis());

    public static int[] generateNumbersArray(int count, int min, int max) {

        int eachNumberRange = Math.round(((float) max) / ((float) count));
        int prevNum = min;
        int maxNum = eachNumberRange;

        int[] numbers = new int[count];
        for(int i = 0; i < count; i++) {
            numbers[i] = generateNumber(prevNum + 1, maxNum);
            prevNum = numbers[i];
            maxNum = prevNum + eachNumberRange;
        }
        return numbers;
    }

    public static int generateNumber(int min, int max) {
        return rand.nextInt((max - min) + 1) + min;
    }


}

File: Android/app/src/main/java/realm/io/realmpop/util/RandomNumberUtils.kt
package realm.io.realmpop.util

import java.util.Random

object RandomNumberUtils {

    private val rand = Random(System.currentTimeMillis())

    fun generateNumbersArray(count: Int, min: Int, max: Int): IntArray {

        val eachNumberRange = Math.round(max.toFloat() / count.toFloat())
        var prevNum = min
        var maxNum = eachNumberRange

        val numbers = IntArray(count)
        for (i in 0..count - 1) {
            if (i == 0) {
                numbers[i] = 0
            } else {
                numbers[i] = generateNumber(prevNum + 1, maxNum)
            }
            prevNum = numbers[i]
            maxNum = prevNum + eachNumberRange
        }
        return numbers
    }

    fun generateNumber(min: Int, max: Int): Int {
        return rand.nextInt(max - min + 1) + min
    }


}
